### PR TITLE
ensure that external hostname is set for stricter servers

### PR DESCRIPTION
**Pull request by loganwright**
#47

### CODE REVIEW

**Review by vzsg — APPROVED** [review]
🦄  🌈

**Review by loganwright — COMMENTED** [review]
I want to verify that this won't affect server implementations 👍

File: Sources/TLS/ClientSocket.swift [hunk 1]
```diff
@@ -37,6 +37,19 @@ extension ClientSocket {
...
         }
 
+        /// https://github.com/vapor/tls/issues/47
+        let hostname = servername ?? socket.hostname
+        var cName = hostname.utf8CString
+        cName.withUnsafeMutableBytes { name in
+            // SSL_set_tlsext_host_name is a C macro,
+            // which is not directly callable in Swift.
+            // This is its expanded form.
+            _ = SSL_ctrl(ssl,
```
Comment (tanner0101) [inline Sources/TLS/ClientSocket.swift]: Xcode formatting

**Review by tanner0101 — COMMENTED** [review]
Looks great! (though a test would also be nice)

**Comment by loganwright** [comment]
@tanner0101 added test 👍

**Pull request merged**